service/src/main/java/bio/terra/catalog/datarepo/DatarepoService.java

```diff
@@ -47,11 +55,18 @@ public DatarepoService(DatarepoConfiguration datarepoConfig) {
     this.commonHttpClient = new ApiClient().getHttpClient();
   }
 
-  public Map<String, List<String>> getSnapshotIdsAndRoles(AuthenticatedUserRequest user) {
+
+  public Map<String, List<DatasetAccessLevel>> getSnapshotIdsAndRoles(AuthenticatedUserRequest user) {
     try {
-      return snapshotsApi(user)
+      Map<String, List<String>> response = snapshotsApi(user)
           .enumerateSnapshots(null, MAX_DATASETS, null, null, null, null, null)
           .getRoleMap();
+      Map<String, List<DatasetAccessLevel>> idsToDatasetAccessLevelList = new java.util.HashMap<>(Map.of());
+      idsToDatasetAccessLevelList.putAll(
+          response.entrySet().stream().collect(Collectors.toMap(
+              Map.Entry::getKey,
+              entry -> entry.getValue().stream().map(SNAPSHOT_ROLE_TO_DATASET_ACCESS_LEVEL::get).toList())));
+      return idsToDatasetAccessLevelList;
```

Comment:
Is this code necessary? I would expect that you could return the result of the streaming operation directly
```suggestion
      return  response.entrySet().stream().collect(Collectors.toMap(
              Map.Entry::getKey,
              entry -> entry.getValue().stream().map(SNAPSHOT_ROLE_TO_DATASET_ACCESS_LEVEL::get).toList());;
```